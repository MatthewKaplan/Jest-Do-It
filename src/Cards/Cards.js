import React from 'react';
import '../Styles/_Cards.scss';
import { AnswerButtons } from '../AnswerButtons/AnswerButtons.js'

export const Cards = (props) => {

  const currentAnswers = props.currentCard.potentialAnswers;
  const currentQuestion = props.currentCard.question;

  const cardAnswers = currentAnswers.map(answer =>
    <AnswerButtons 
      key={answer}
      answer={answer}
    />)
  
  return (
    <article className="questionCard">
      <section className="cardQuestion">
        <h3>{currentQuestion}</h3>
      </section>
      <section className="cardAnswers">
        {cardAnswers}
      </section>
    </article>
  )
}

export default Cards;