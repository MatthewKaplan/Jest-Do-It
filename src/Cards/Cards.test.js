import React from 'react';
import  Cards  from './Cards';
import { shallow } from 'enzyme';

const mockCurrentCard = 
{
id: 3,
question: "When you want full DOM Rendering what must you import from the Enzyme library at the top of your test file?",
potentialAnswers: [
  "mount",
  "shallow",
  "sinon",
  "expect"
],
correctAnswer: "mount",
save: "false",
onCorrectAnswer: "Thats correct! For full DOM rendering you'll want to use mount. If you just want to render a shallow copy you'd use shallow.",
onIncorrectAnswer: "Thats incorrect. The correct answer is: mount.",
linkName: "Enzyme mount",
link: "https://airbnb.io/enzyme/#full-dom-rendering"
}


const mock_nextCard = jest.fn()

describe("Cards", () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Cards 
        currentCard={mockCurrentCard}
        nextCard={mock_nextCard}
      />
    );
  });

  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });

  // it("should show the next card when clicked", () => {
  //   wrapper.find("[data-test='next-card-btn']").simulate('click', { preventDefault: () => {} });
  //   expect(mock_nextCard).toBeCalled();
  // });
});









