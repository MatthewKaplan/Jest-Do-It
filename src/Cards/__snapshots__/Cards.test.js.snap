// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should match the snapshot with all data passed in 1`] = `
<article
  className="questionCard"
>
  <section
    className="cardQuestion"
  >
    <h3>
      When you want full DOM Rendering what must you import from the Enzyme library at the top of your test file?
    </h3>
  </section>
  <section
    className="cardAnswers"
  >
    <AnswerButtons
      answer="mount"
      key="mount"
    />
    <AnswerButtons
      answer="shallow"
      key="shallow"
    />
    <AnswerButtons
      answer="sinon"
      key="sinon"
    />
    <AnswerButtons
      answer="expect"
      key="expect"
    />
  </section>
  <section
    className="answerResponse"
  >
    <h3 />
  </section>
</article>
`;
